src/ServiceControl.Infrastructure.RavenDB/Extensions.cs

```diff
@@ -16,5 +19,48 @@ public static void Query<TState>(this DocumentDatabase db, string index, IndexQu
                 onItem(doc, state);
             }
         }
+
+        public static void ThrowWhenIndexErrors(this EmbeddableDocumentStore documentStore)
+        {
+            var statistics = documentStore.DatabaseCommands.GetStatistics();
+
+            if (statistics.Errors.Length > 0)
+            {
+                var text = new StringBuilder();
+                text.AppendLine("Detected RavenDB index errors, please start maintenance mode and resolve the following issues:");
+                foreach (var indexError in statistics.Errors)
+                {
+                    text.AppendLine($"- Index [{indexError.IndexName}] error: {indexError.Error} (Action: {indexError.Action},  Doc: {indexError.Document}, At: {indexError.Timestamp})");
+                }
+                throw new Exception(text.ToString());
```

Comment:
should we consider creating a specific exception? Something like "RavenDBHasStaleIndexes"?